site/en/docs/extensions/mv3/tut_puppeteer-testing/index.md

```diff
@@ -0,0 +1,141 @@
+---
+layout: "layouts/doc-post.njk"
+title: "Tutorial: Testing Chrome Extensions with Puppeteer"
+seoTitle: "Tutorial: Testing Chrome Extensions with Puppeteer"
+date: 2023-10-09
+description: How to write an automated test for Chrome Etensions in Puppeteer.
+---
+
+_For general advice on testing extensions, see
+[Automated testing for Chrome Extensions][automated-testing]._
+
+[Puppeteer][puppeteer] provides a framework for building automated tests of websites, which also
+includes the ability to test Chrome Extensions. These are high-level integration tests that test the
+functionality of a website or extension once it has been built into the final product.
+
+## Before you start
+
+Clone or download the [chrome-extensions-samples][samples-repo] repository. We'll use the history
+API demo in `api-samples/history/showHistory` as our test extension.
+
+You'll also need to install [Node.JS][node] which is the runtime Puppeteer is built on.
+
+## Writing your test
+
+### Step 1: Start your Node.JS project
+
+We need to set up a basic Node.JS project. In a new folder, create a `package.json` file with the
+following:
+
+```js
+{
+  "name": "puppeteer-demo",
+  "version": "1.0"
+}
+```
+
+Similar to an extension's manifest.json file, this file is required by all Node projects.
+
+### Step 2: Install Puppeteer
+
+Run `npm install puppeteer` to add Puppeteer as a dependency. It will be automatically added to your
+`package.json` file.
+
+### Step 3: Create an entry point
+
+In a new file called `index.js`, add the following:
+
+```js
+const puppeteer = require('puppeteer');
+
+const EXTENSION_PATH = '../../api-samples/history/showHistory';
+const EXTENSION_ID = 'jkomgjfbbjocikdmilgaehbfpllalmia';
+
+async function run() {
+  const browser = await puppeteer.launch({
+    headless: false,
+    args: [
+      `--disable-extensions-except=${EXTENSION_PATH}`,
+      `--load-extension=${EXTENSION_PATH}`
+    ]
+  });
+}
+
+run();
+```
+
+Run `node index.js` and you should see Chrome open with your extension loaded.
+
+### Step 4: Add an alias
+
+To make running the tests easier, add an alias to your `package.json` file:
+
+```js
+{
+  "name": "puppeteer-demo",
+  "version": "1.0",
+  "dependencies": {
+    "puppeteer": "^21.3.6"
+  },
+  "scripts": {
+    "start": "node index.js"
```

Comment:
Why not `test`?

Reply:
Just personal preference here, I can change it if you feel strongly. Since this whole NPM project **is** the test I think having `start` run the tests makes sense. I usually think of `test` as testing the NPM project, which would in this case mean testing the tests.

Reply:
No strong preferences.  It makes it easier to always use the same task name because it makes test execution easier in a multi-project setup.